source/transactions-convenient-api/tests/unified/callback-retry.yml

```diff
@@ -0,0 +1,245 @@
+                readConcern: { $$exists: false }
+                startTransaction: { $$exists: false }
+                writeConcern: { $$exists: false }
+              commandName: abortTransaction
+              databaseName: admin
+          - commandStartedEvent:
+              command:
+                insert: *collectionName
+                documents:
+                  - { _id: 1 }
+                ordered: true
+                lsid: { $$sessionLsid: *session0 }
+                # second transaction will be causally consistent with the first
+                readConcern: { afterClusterTime: { $$exists: true } }
+                # txnNumber is incremented when retrying the transaction
+                txnNumber: { $numberLong: "2" }
+                startTransaction: true
+                autocommit: false
+                # omitted fields
+                writeConcern: { $$exists: false }
+              commandName: insert
+              databaseName: *databaseName
+          - commandStartedEvent:
+              command:
+                abortTransaction: 1
+                lsid: { $$sessionLsid: *session0 }
+                txnNumber: { $numberLong: "2" }
+                autocommit: false
+                # omitted fields
+                readConcern: { $$exists: false }
+                startTransaction: { $$exists: false }
+                writeConcern: { $$exists: false }
+              commandName: abortTransaction
+              databaseName: admin
+          - commandStartedEvent:
+              command:
+                insert: *collectionName
+                documents:
+                  - { _id: 1 }
+                ordered: true
+                lsid: { $$sessionLsid: *session0 }
+                # third transaction will be causally consistent with the second
+                readConcern: { afterClusterTime: { $$exists: true } }
+                # txnNumber is incremented when retrying the transaction
+                txnNumber: { $numberLong: "3" }
+                startTransaction: true
+                autocommit: false
+                # omitted fields
+                writeConcern: { $$exists: false }
+              commandName: insert
+              databaseName: *databaseName
+          - commandStartedEvent:
+              command:
+                commitTransaction: 1
+                lsid: { $$sessionLsid: *session0 }
+                txnNumber: { $numberLong: "3" }
+                autocommit: false
+                # omitted fields
+                readConcern: { $$exists: false }
+                startTransaction: { $$exists: false }
+                writeConcern: { $$exists: false }
+              commandName: commitTransaction
+              databaseName: admin
+    outcome:
+      - collectionName: *collectionName
+        databaseName: *databaseName
+        documents:
+          - { _id: 1 }
+  -
+    description: callback is not retried after non-transient error (DuplicateKeyError)
```

Comment:
The legacy test has useMultipleMongoses: true for this one, and the Java driver's legacy test runner skip it when running against a replica set. I'm not sure whether that is correct behavior, but when running this test as a UTR it fails against a 7.0 replica set (it actually times out on Evergreen after about 40 minutes). Can we determine whether it's intended to be run against a replica set?

Reply:
It also times out against a sharded cluster, whereas the legacy test passes.

Reply:
If this test is passing for others than it's likely an issue with the Java driver UTR.

Reply:
I think what I'm running into here is https://jira.mongodb.org/browse/DRIVERS-2816

Reply:
@jyemin: In response to your [top-level comment](https://github.com/mongodb/specifications/pull/1505#issuecomment-1941807228) above:

> There are four other tests failing as well, but those seem to be test runner issues.
>
> Patch build: https://spruce.mongodb.com/version/65cb8c513e8e86ed75242d71/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

I see that two tasks timed out, but don't see any failures within them. Are you referring to something in a different patch build?

Reply:
The test that runs for 40 minutes on Evergreen before the Evergreen task times out fails when run locally after about a minute.  So I was able to see these four other test failures locally but not yet on Evergreen.